client/my-sites/plans/current-plan/main.jsx

```diff
@@ -32,12 +36,42 @@ const PlanDetailsComponent = React.createClass( {
 		sitePlans: React.PropTypes.object.isRequired,
 		fetchPlans: React.PropTypes.func.isRequired
 	},
+
 	componentWillUpdate: function( props ) {
 		this.props.fetchPlans( props );
 	},
+
 	componentDidMount: function() {
 		this.props.fetchPlans();
 	},
+
+	getPurchaseInfo: function() {
+		const plan = this.props.currentPlan;
+
+		if ( ! plan ) {
+			return null;
+		}
+
+		const classes = classNames( 'current-plan__purchase-info', {
+			'is-expiring': plan.userFacingExpiryMoment < this.moment().add( 30, 'days' )
```

Comment:
We'll probably want to move toward letting a selector populate a moment instance, to allow for plan persistence in redux.